code/src/containers/FilterLink.js

```diff
@@ -0,0 +1,16 @@
+import { connect } from 'react-redux'
+import { setVisibilityFilter } from '../actions'
+import Link from '../components/Link'
+
+const mapStateToProps = (state, ownProps) => ({
+  active: ownProps.filter === state.visibilityFilter
+})
+
+const mapDispatchToProps = (dispatch, ownProps) => ({
+  onClick: () => dispatch(setVisibilityFilter(ownProps.filter))
+})
```

Comment:
mapStateToProps and mapDispatchToProps is from before we had React Hooks.
Have a[ read here](https://blog.bitsrc.io/using-react-redux-hooks-97654aff01e4) regarding this